Chore: Update Auth0

Spazzy757 opened the pull request:
# Changelog

- `github.com/auth0/go-jwt-middleware/validate/josev2` has been removed from the auth0 library in favour of `github.com/auth0/go-jwt-middleware/v2/validator`
- updated local dev environment (I added a Makefile in order to make things slightly easier to get started with)
- added documentation to help with getting started and development
- added unit testing for the Auth Middleware

FIX: https://github.com/vlab-research/vlab/issues/105

Spazzy757 commented:
@nandanrao All tests seem to pass and it seems to run without issue. Let me know your thoughts on structure and docs

nandanrao commented on dashboard-api/internal/platform/server/middleware/auth/auth_test.go:
```diff
@@ -0,0 +1,144 @@
+// SetupTestJWKServer creates a fake server to respond to JWK requests
+// TODO: Move to internal/testhelpers, currently not possible due to cycle
+// import
+func SetupTestJWKServer(
```
This seems like a great thing to have for tests, makes it all very easy to test fully!

Spazzy757 replied:
This was taken from Auth0's library so cant take all the credit

nandanrao commented on dashboard-api/Makefile:
```diff
@@ -0,0 +1,60 @@
+default: help
```
Love having a Makefile. 

Seems like there are two versions, one where we use docker compose and one where we just run `go test`. 

The docker compose, I assume, is for running the DB? But `go test` doesn't need the DB, or...?

Spazzy757 replied:
Yes this was some confusion for me as well as you have a `docker-compose.test.yaml`, but you don't actually need a test. I added the target as it was in the test.sh file

Spazzy757 replied:
Let me know if you want me to delete the docker-compose stuff

nandanrao replied:
Yeah, if it's not being used, clean it out!

Spazzy757 replied:
Did a clean up of unnecessary files as well.

nandanrao commented on dashboard-api/internal/platform/server/middleware/auth/auth_test.go:
```diff
@@ -0,0 +1,144 @@
+	"gopkg.in/square/go-jose.v2/jwt"
+)
+
+func TestEnsureValidTokenMiddleware(t *testing.T) {
```
Very reasonable!!

nandanrao commented on dashboard-api/internal/platform/server/middleware/auth/auth_test.go:
```diff
@@ -0,0 +1,144 @@
+	testServer := SetupTestJWKServer(t, jwk)
+	defer testServer.Close()
+
+	t.Run("invalid token", func(t *testing.T) {
```
I would make the name a bit more descriptive so we know _why_ the token is invalid (i.e. the audience is wrong)

Spazzy757 replied:
:100: will update it

Spazzy757 commented on .github/workflows/dashboard-api-tests.yml:
```diff
@@ -1,4 +1,4 @@
-name: Dashboard Api Tests
+name: Dashboard Api
```
Changed this because its redundant, i.e it will now be `Dashboard Api / Tests` instead of `Dashboard Api Tests / Test`

Spazzy757 commented:
Also @nandanrao I removed looper (its been archived) Im not sure why we were using it (looks like the tests reran every time you made changes), but you can do hot reloads using air (I've updated the config to watch go files) Let me know if this fits

nandanrao commented:
Sounds good. I think looper was there to do TDD - it's what I always use locally - really just an alternative for `watch go test`. Although, to be honest, not sure what advantage is offers. 

But ok - it all looks good to me!
